[frontent-api] GQL resolvers refactoring (#2563)
shopsys/shopsys: upgrade/UPGRADE-v11.0.0-dev.md

    - also see #project-base-diff for more information about changes needed to be done in your project
- lock version of npm `jquery-ui` to version `1.12.1` in order to fix category sorting in admin ([#2558](https://github.com/shopsys/shopsys/pull/2558))
    - see #project-base-diff for more information about changes needed to be done in your project
- GQL resolvers and mutations refactoring ([#2563](https://github.com/shopsys/shopsys/pull/2563))

>     - all GQL Resolvers are renamed to Query:

Author: sure

>         - `Shopsys\FrontendApiBundle\Model\Resolver\Advert\AdverPositionsResolver` class are renamed:

Author: ok

        ```
        - `Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductsResolver` class are renamed:
        ```diff
        - Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductsResolver
        + Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductsQuery
        ```
        - `Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductDetailResolver` class are renamed:
        ```diff
        - Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductDetailResolver
        + Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductDetailQuery
        ```
        - `Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportsResolver` class are renamed:
        ```diff
        - Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportsResolver
        + Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportsQuery
        ```
        - `Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportResolver` class are renamed:
        ```diff
        - Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportResolver
        + Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportQuery
        ```

>     - All functions in Query classes used as alias are renamed by convention with suffix `Query`

Author: ok

>         - in `Shopsys\FrontendApiBundle\Model\Resolver\Advert\AdverPositionsQuery` class is renamed function:

Author: ok

        ```diff
        - public function resolve(): array
        + public function advertPositionsQuery(): array
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Advert\AdversQuery` class in renamed function:
        ```diff
        - public function resolve(?string $positionName = null): array
        + public function advertsQuery(?string $positionName = null): array
        ```

>         - in `Shopsys\FrontendApiBundle\Model\Resolver\Article\ArticleQuery` class are renamed funcitons:

Author: ok

        - in `Shopsys\FrontendApiBundle\Model\Resolver\Category\CategoryQuery` class is renamed function:
        ```diff
        - public function resolveByUuidOrUrlSlug(?string $uuid = null, ?string $urlSlug = null): Category
        + public function categoryByUuidOrUrlSlugQuery(?string $uuid = null, ?string $urlSlug = null): Category
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Category\CategoriesQuery` class is renamed function:
        ```diff
        - public function resolve(): array
        + public function categoriesQuery(): array
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Category\CategoriesSearchQuery` class is renamed function:
        ```diff
        - public function resolveSearch(Argument $argument)
        + public function categoriesSearchQuery(Argument $argument)
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\User\CurrentCustomerUserQuery` class is renamed function:
        ```diff
        - public function resolver(): CustomerUser
        + public function currentCustomerUserQuery(): CustomerUser
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Image\ImagesQuery` class are:

>           - replaced functions `resolveByCategory`, `resolveByPayment`, `resolveByTransport`, `resolveByBrand` for `imagesByEntityQuery`

Author: ok

        + public function productsByCategoryQuery(Argument $argument, Category $category)
        ```
        ```diff
        - public function resolveByBrand(Argument $argument, Brand $brand)
        + public function productsByBrandQuery(Argument $argument, Brand $brand)
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Products\ProductDetailQuery` class is renamed function:
        ```diff
        - public function resolver(?string $uuid = null, ?string $urlSlug = null): array
        + public function productDetailQuery(?string $uuid = null, ?string $urlSlug = null): array
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportsQuery` class is renamed function:
        ```diff
        - public function resolve(): array
        + public function transportsQuery(): array
        ```
        - in `Shopsys\FrontendApiBundle\Model\Resolver\Transport\TransportQuery` class is renamed function:
        ```diff
        - public function resolver(string $uuid): Transport
        + public function transportByTransportUuidQuery(string $uuid): Transport
        ```

>     - in all GQL queries and attribute resolve definitions are replaced all old aliases for renamed Query functions, e.g.:

Reviewer: ```suggestion
    - in all GQL queries resolve definitions are all old aliases replaced by renamed Query functions, e.g.:
```

Author: ok

        ```diff
        - resolve: '@=query("Shopsys\\FrontendApiBundle\\Model\\Resolver\\Image\\ImagesResolver::resolveByAdvert", value, args["type"], args["size"])'
        + resolve: '@=query("imagesByAdvertQuery", value, args["type"], args["size"])'
        ```

>     - item in images array cannot be null and array as it is not nullable too. The correction is concerned `AdvertImageDecorator`, `BrandDecorator` and `CategoryDecorator`.

Reviewer: ```suggestion
    - type definition of `images` in `AdvertImageDecorator`, `BrandDecorator` and `CategoryDecorator` has changed:
```

Author: ok

        ```diff
        - type: "[Image]"
        + type: "[Image!]!"
        ```

>     - in GQL definition of `AdvertImageDecorator` is fixed attribute for images array.

Reviewer: ```suggestion
    - in GQL definition of `AdvertImageDecorator` is attribute `image` renamed to `images` and its type definition has changed:
```

Author: oki

        ```diff
        - image:
        -     type: "[Image]"
        + images:
        +     type: "[Image!]!"
        ```

>     - All functions in Mutation classes used as alias are renamed by convention with suffix `Mutation`

Reviewer: ```suggestion
    - All mutation methods in `*Mutation` classes should end with suffix `Mutation` in order to be automatically aliased for use in GQL definitions
```

Author: ok

        ```
        - in `Shopsys\FrontendApiBundle\Model\Mutation\Logout\LogoutMutation` class is renamed function:
        ```diff
        - public function logout(): array
        + public function logoutMutation(): array
        ```
        - in `Shopsys\FrontendApiBundle\Model\Mutation\Newsletter\NewsletterMutation` class is renamed function:
        ```diff
        - public function newsletterSubscribe(Argument $argument, InputValidator $validator): array
        + public function newsletterSubscribeMutation(Argument $argument, InputValidator $validator): array
        ```
        - in `Shopsys\FrontendApiBundle\Model\Mutation\Order\CreateOrderMutation` class is renamed function:
        ```diff
        - public function createOrder(Argument $argument, InputValidator $validator): Order
        + public function createOrderMutation(Argument $argument, InputValidator $validator): Order
        ```
  - in GQL mutation definitions are replaced all old aliases for renamed Mutation functions, e.g.:
        ```diff
        - resolve: "@=mutation('create_order', args, validator)"
        + resolve: "@=mutation('createOrderMutation', args, validator)"

>         ```

Reviewer: As I see this, there is no simple rule to be applied. You have changed names in multiple ways thus I think you should create list of changes

Author: ok, i added all changes
